src/main/java/bio/terra/janitor/service/janitor/TrackedResourceService.java

```diff
@@ -116,60 +108,47 @@ private TrackedResourceId createResourceAndUpdateDuplicates(
+    return resourceAndLabels.map(TrackedResourceService::createInfo);
   }
 
   /** Retrieves the resources with the {@link CloudResourceUid}. */
-  public TrackedResourceInfoList getResources(
-      CloudResourceUid cloudResourceUid, AuthenticatedUserRequest userReq) {
-    iamService.requireAdminUser(userReq);
+  public TrackedResourceInfoList getResources(CloudResourceUid cloudResourceUid) {
     List<TrackedResourceAndLabels> resourcesWithLabels =
         janitorDao.retrieveResourcesWith(cloudResourceUid);
     TrackedResourceInfoList resourceList = new TrackedResourceInfoList();
     resourcesWithLabels.stream()
-        .map(JanitorService::createInfo)
+        .map(TrackedResourceService::createInfo)
         .forEach(resourceList::addResourcesItem);
     return resourceList;
   }
 
+  /** Updates the READY resource state to ABANDONED. */
+  public void abandonResource(CloudResourceUid cloudResourceUid) {
+    List<TrackedResource> resources =
+        getResourceWithState(
+            cloudResourceUid, TrackedResourceState.READY, TrackedResourceState.CLEANING);
+
+    if (resources.size() > 1) {
+      throw new InternalServerErrorException(
+          "More than one READY or CLEANING state resources are found.");
+    }
+    janitorDao.updateResourceState(
+        resources.get(0).trackedResourceId(), TrackedResourceState.ABANDONED);
+  }
+
   /**
-   * Updates the resource state.
-   *
-   * <p>Currently it supports:
+   * Updates the ABANDONED resource state to READY.
    *
-   * <ul>
-   *   <li>Abandon resource: Update resource from READY or CLEANING to ABANDONED
-   *   <li>Bump resource: Update resource from ABANDONED to READY
-   * </ul>
+   * <p>It is possible that there might be multiple ABANDON resources, and we only need to update
+   * the one with last expiration time.
    */
-  public void updateResource(
-      CloudResourceUid cloudResourceUid, ResourceState state, AuthenticatedUserRequest userReq) {
-    iamService.requireAdminUser(userReq);
-    if (state == ResourceState.ABANDONED) {
-      // Update state to ABANDONED for READY or CLEANING state resources.
-      List<TrackedResource> resources =
-          getResourceWithState(
-              cloudResourceUid, TrackedResourceState.READY, TrackedResourceState.CLEANING);
-
-      if (resources.size() > 1) {
-        throw new InternalServerErrorException(
-            "More than one READY or CLEANING state resources are found.");
-      }
-      janitorDao.updateResourceState(
-          resources.get(0).trackedResourceId(), TrackedResourceState.ABANDONED);
-    } else if (state == ResourceState.READY) {
-      // Bump ABANDONED state resources for cleaning.
-      List<TrackedResource> resources =
-          getResourceWithState(cloudResourceUid, TrackedResourceState.ABANDONED);
+  public void bumpResource(CloudResourceUid cloudResourceUid) {
+    List<TrackedResource> resources =
+        getResourceWithState(cloudResourceUid, TrackedResourceState.ABANDONED);
```

Comment:
We should also be able to bump ERROR resources, not just ABANDONED. also update comment.

Reply:
DONE